### ISSUE TITLE
Twitter report: Warn others does not work "Es konnte keine Übermittlungs-TAN erstellt werden."

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#4815]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Pixel 4a
- Android version: 12
- App version: 2.17.2

## Describe the bug

The Twitter user requested a TAN via the TAN hotline, entered it in the app and stepped through the "Warn others" process. However, at the very end, he sees this error message:
<img src="https://user-images.githubusercontent.com/67682506/153267436-1b81efc7-0433-4edf-ba32-246203df8bda.jpeg" width="300">

## Steps to reproduce the issue

1. Request a TAN at the hotline
2. Enter the TAN
3. Step through the Warn others process
4. See the error at the end of the process

## Expected behaviour

No error.

## Additional context

Clicking on "Details" reveals this stack trace:
```
Ursache:
de.rki.coronawarnapp.exception.TanPairingException: Tan has been retrieved before for this registration token
	at de.rki.coronawarnapp.playbook.DefaultPlaybook.submit(DefaultPlaybook.kt:29)
	at de.rki.coronawarnapp.playbook.DefaultPlaybook$submit$1.invokeSuspend(Unknown Source:12)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:4)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:18)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:10)
Caused by: de.rki.coronawarnapp.exception.http.BadRequestException: Error during web request: code=400 message= body='
	at de.rki.coronawarnapp.http.HttpErrorParser.intercept(HttpErrorParser.kt:36)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:14)
	at de.rki.coronawarnapp.http.interceptor.RetryInterceptor.intercept(RetryInterceptor.kt:3)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:14)
	at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.kt:5)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:14)
	at de.rki.coronawarnapp.http.interceptor.WebSecurityVerificationInterceptor.intercept(WebSecurityVerificationInterceptor.kt:3)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:14)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:25)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:12)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:920)
```

### This is a Twitter report
The information in this report is based on the feedback from a Twitter user: https://twitter.com/kaugummiheld/status/1491100030589345792?s=20&t=4awuiciEbF56ti451dKT6Q. He said the TAN was entered into the app 2 minutes after he got it from the hotline. 

---
Internal Tracking ID: [EXPOSUREAPP-11782](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11782)

**Comment by vaubaehn** [comment]
It looks to be the same problem, that has already been reported one year ago. Because at that time the error dialog wasn't specific enough, the dialog was enhanced in https://github.com/corona-warn-app/cwa-app-android/pull/2161 .
What I understood from @mtwalli 's explanation is, that a user got a TAN from hotline, but when CWA tried to retrieve a token for that into CWA entered TAN from the web server, the web server couldn't find such a token.
It's unclear why the token can't be retrieved, but it looks like, CWA can't do anything about it.

Personally I would think it would be worth to try again after 5 or 10 minutes (maybe there is just a delay in the server infrastructure), but if it fails again, a new TAN would need to be generated by the hotline.

**Comment by svengabr** [comment]
Internal Tracking ID: EXPOSUREAPP-11782

**Comment by Ein-Tim** [comment]
@vaubaehn 

He tried it multiple times on different dates. Should I suggest to get a new TAN?

**Comment by vaubaehn** [comment]
@Ein-Tim 
In that case, I recommended to do so.
Hope there's not much waiting time in the hotline...

**Comment by vaubaehn** [comment]
@dsarkar @mtwalli 
The dialog text recommends to call the technical helpline... What is supposed how they can help in that case?

**Comment by Ein-Tim** [comment]
@vaubaehn 

Okay, I suggested to call the hotline again. Is there really nothing what the app can do in this case?

**Comment by vaubaehn** [comment]
> @vaubaehn
> 
> Okay, I suggested to call the hotline again. Is there really nothing what the app can do in this case?

@Ein-Tim 
I don't think that anything on CWA side can be done technically (although it's worth to re-evaluate the dialog text/whether the technical help line can actually help), but I would forward this question to @mtwalli to be sure.

**Comment by vaubaehn** [comment]
@dsarkar 
Do you know wether there is a plausibility-check of the entered TAN-code (like a one-digit 'Prüfsumme' inside the TAN-code itself) before CWA trying to retrieve the corresponding token on the server? If there is no such check, a simple typo (or misunderstood letter/number when read by the call center agent to the user) could cause this error, couldn't it?

**Comment by Ein-Tim** [comment]
@vaubaehn 

Probably, all what the technical hotline can do to help you, is redirect you to the verification hotline.

**Comment by citizenserious** [comment]
Cause:
de.rki.coronawarnapp.exception.TanPairingException: Tan has been retrieved before for this registration token
	at de.rki.coronawarnapp.playbook.DefaultPlaybook.wrapException(DefaultPlaybook.kt:131)
	at de.rki.coronawarnapp.playbook.DefaultPlaybook.submit(DefaultPlaybook.kt:114)
	at de.rki.coronawarnapp.playbook.DefaultPlaybook$submit$1.invokeSuspend(Unknown Source:15)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: de.rki.coronawarnapp.exception.http.BadRequestException: Error during web request: code=400 message= body='
	at de.rki.coronawarnapp.http.HttpErrorParser.intercept(HttpErrorParser.kt:80)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at de.rki.coronawarnapp.http.interceptor.RetryInterceptor.intercept(RetryInterceptor.kt:15)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.kt:154)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at de.rki.coronawarnapp.http.interceptor.WebSecurityVerificationInterceptor.intercept(WebSecurityVerificationInterceptor.kt:11)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)

**Comment by Ein-Tim** [comment]
@svengabr @dsarkar @thomasaugsten @larswmh 

Do we have an incident here?

Also see a Twitter user reporting this: https://twitter.com/timm_1991/status/1504100365171634179?s=21

FYI @fynngodau

**Comment by citizenserious** [comment]
> @vaubaehn
> 
> Probably, all what the technical hotline can do to help you, is redirect you to the verification hotline.

After 20 minutes waiting in the hotline someone picked up and immediately hang up.

**Comment by Ein-Tim** [comment]
Another Twitter user having the same problem: https://twitter.com/_derTheoretiker/status/1514536930494623750?s=20&t=KQceXEF8eDkHhoJylWdcPg

He already tried it with separate TANs issued from the hotline but always receives this error. I asked him for an error log.

@thomasaugsten FYI

**Comment by Ein-Tim** [comment]
Here's the error log: 
[CWA Log 2022-04-14 18_33_24.372.txt](https://github.com/corona-warn-app/cwa-app-android/files/8491382/CWA.Log.2022-04-14.18_33_24.372.txt)

**Comment by Ein-Tim** [comment]
@mlenkeit Hey Maximilian, just a friendly reminder, maybe you or the team could take a look at the error log I shared above.

**Comment by dsarkar** [comment]
@Ein-Tim error log forwarded to internal ticket. Thanks!

**Comment by mlenkeit** [comment]
@Ein-Tim thanks for sharing the log! I'm afraid the timeframe that is covered by the log is too short. We would need a log that captures what a sufficient amount of time before the error occurs. Ideally, the entire time frame from when the positive test result is received until the errors occurs. Realistically, this means users would need to turn error logging on before registering a test.

**Comment by Ein-Tim** [comment]
Here's another report on this: #5248.

The strack trace @FrederikP shared via https://github.com/corona-warn-app/cwa-app-android/issues/5248#issuecomment-1142016804 looks very similar to the one I posted in the OP.

**Comment by dsarkar** [comment]
@Ein-Tim  @FrederikP Thanks, case linked in internal ticket.

**Comment by Ein-Tim** [comment]
@mlenkeit 

This seems to be a quite common problem: 
- https://twitter.com/count_tofu/status/1534923746359771138?s=21&t=rmW6bZnxXiCpYFICMUE1bg
- https://twitter.com/wasserspeierdd1/status/1534930834326704128?s=21&t=rmW6bZnxXiCpYFICMUE1bg

**Comment by mlenkeit** [comment]
Here are some related PRs for cwa-server:
- https://github.com/corona-warn-app/cwa-server/pull/1849
- https://github.com/corona-warn-app/cwa-server/pull/1854

**Comment by mlenkeit** [comment]
> Did the app already warn others or should I call the TAN hotline to get a TAN?

@Ein-Tim it depends on the details of the _missing internet connection_ error message. From this information it's difficult to tell.

With a TAN from the TAN hotline, we would ensure that the warning is shared (given that the upload goes through then).

**Comment by Ein-Tim** [comment]
@mlenkeit 

The stack trace of the missing internet connection message is lost, sadly.

I'll tell the user to delete the old test and request a TAN at the hotline, just to be on the safe side.

Thanks for your fast response.

**Comment by dsarkar** [comment]
FYI This PR is intended to improve the situation: https://github.com/corona-warn-app/cwa-app-android/pull/5335

**Comment by mlenkeit** [comment]
Another PR somewhat related to this problem is #5384

**Comment by mlenkeit** [comment]
We hope that https://github.com/corona-warn-app/cwa-app-android/pull/5335 fixes this issue for good. The fix will most likely be included in 2.26.

**Comment by Ein-Tim** [comment]
I can't verify the fix myself bug I assume the bug was fixed successfully.
